src/client/src/tscSQLParser.c

```diff
@@ -5034,6 +5040,21 @@ int32_t handleNeOptr(tSqlExpr** rexpr, tSqlExpr* expr) {
   return TSDB_CODE_SUCCESS;
 }
 
+static void convertWhereStringCharset(tSqlExpr* pRight){
+  if(pRight->value.nType != TSDB_DATA_TYPE_BINARY){
+    return;
+  }
+
+  char newData[TSDB_MAX_NCHAR_LEN] = {0};
```

Comment:
big allocation on stack. the max length is pRight->value.nLen, malloc it then free?